diesel/src/sqlite/connection/mod.rs

```diff
@@ -372,4 +372,105 @@ mod tests {
             .get_result::<(i32, i32, i32)>(&connection);
         assert_eq!(Ok((2, 3, 4)), added);
     }
+
+    use crate::sql_types::Foldable;
+    sql_function! {
+        #[aggregate]
+        fn my_sum<ST: Foldable>(expr: ST) -> ST::Sum;
+    }
+
+    struct MySum<T: Foldable> {
+        sum: T::Sum,
+    }
+    impl<T: Foldable> MySum<T> {
+        fn new() -> Self {
+            Self { sum: 0 }
+        }
+    }
+    impl<T: Foldable> Aggregator<T> for MySum<T> {
+        fn step(&mut self, expr: T) {
+            self.sum += expr;
+        }
+
+        fn finalize(&self) -> T::Sum {
+            self.sum
+        }
+    }
+    table! {
+        my_sum_example {
+            id -> Integer,
+            value -> Integer,
+        }
+    }
+
+    #[test]
+    fn register_aggregate_function() {
+        use self::my_sum_example::dsl::*;
+
+        let connection = SqliteConnection::establish(":memory:").unwrap();
+        connection.execute("CREATE TABLE my_sum_example (id integer primary key autoincrement, value integer)").unwrap();
+        connection.execute("INSERT INTO my_sum_example (value) VALUES (1), (2), (3)").unwrap();
+
+        let my_sum_struct = MySum::new();
+        my_sum::register_impl(&connection, my_sum_struct).unwrap();
```

Comment:
Registering a custom aggregate function will need to only pass the corresponding type, not a specific instance, because otherwise you would use the same instance for all calls to that function, which would quite likely corrupt the results, due to invalid states.

So something like `my_sum::register_impl::<MySum>(&connection)`